### PR TITLE
test: batch check unit tests

### PR DESCRIPTION
**Pull request by justincoh**
## Description

This PR is all unit tests for #2039. For integration tests, see its sister PR #2066

## References

This PR is one of two created by splitting #2047 into two. The other one is batch check integration tests in #2066

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

If you haven't done so yet, we would appreciate it if you could star the [OpenFGA repository](https://github.com/openfga/openfga). :)

### CODE REVIEW

File: pkg/server/commands/batch_check_command_test.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+			Times(numChecks).
+			Return(nil, nil)
+
+		params := &BatchCheckCommandParams{
+			AuthorizationModelID: ts.GetAuthorizationModelID(),
+			Checks:               checks,
+			StoreID:              ulid.Make().String(),
+		}
+
+		result, _, err := cmd.Execute(context.Background(), params)
```
Comment (willvedd) [inline pkg/server/commands/batch_check_command_test.go:72]: **Question:** Are there any assertions that can be made against metadata?

File: pkg/server/commands/batch_check_command_test.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+		params := &BatchCheckCommandParams{
+			AuthorizationModelID: ts.GetAuthorizationModelID(),
+			Checks:               checks,
+			StoreID:              ulid.Make().String(),
+		}
+
+		_, _, err := cmd.Execute(context.Background(), params)
+
+		var expectedErr *BatchCheckValidationError
```
Comment (willvedd) [inline pkg/server/commands/batch_check_command_test.go:192]: **Suggestion:** Make this more specific to the empty string error.

File: pkg/server/commands/batch_check_command_test.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+		// create context and cancel immediately
+		ctx := context.Background()
+		ctx, cancel := context.WithCancel(ctx)
+		cancel()
+
+		response, _, err := cmd.Execute(ctx, params)
+		require.NoError(t, err) // request itself should not error
+
+		// But all checks should return a context cancelled error
+		for _, v := range response {
```
Comment (willvedd) [inline pkg/server/commands/batch_check_command_test.go:225]: **Request: ** Can you also assert the number of result keys equals `numChecks`?

File: pkg/server/batch_check_test.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+	"github.com/openfga/openfga/internal/condition"
+	"github.com/openfga/openfga/internal/graph"
+	"github.com/openfga/openfga/pkg/server/commands"
+)
+
+func TestBatchCheckUsesTypesystemModel(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
```
Comment (willvedd) [inline pkg/server/batch_check_test.go:25]: **Request:** Add this to all your unit tests.

File: pkg/server/commands/batch_check_command_test.go (hunk 1 above)
Comment (justincoh) [reply pkg/server/commands/batch_check_command_test.go:78]: The only piece of metadata is the query count which is `0` in the unit test runner, but I can at least assert that the key and value are in there to catch and possible future modifications

File: pkg/server/commands/batch_check_command_test.go (hunk 2 above)
Comment (justincoh) [reply pkg/server/commands/batch_check_command_test.go:202]: Added an additional assertion below on the specific message. Or did you mean the error type itself?

**Review by willvedd — APPROVED** [review]

**Pull request merged**